pkg/domain/domain.go

```diff
@@ -916,19 +916,17 @@ func (do *Domain) topologySyncerKeeper() {
 
 func (do *Domain) refreshMDLCheckTableInfo() {
 	se, err := do.sysSessionPool.Get()
-
 	if err != nil {
 		logutil.BgLogger().Warn("get system session failed", zap.Error(err))
 		return
 	}
+	defer do.sysSessionPool.Put(se)
 	// Make sure the session is new.
 	sctx := se.(sessionctx.Context)
 	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnMeta)
 	if _, err := sctx.GetSQLExecutor().ExecuteInternal(ctx, "rollback"); err != nil {
-		se.Close()
```

Comment:
May I ask why do we need to drop this line?

Reply:
Good catch. It seems the behavior changed. I have reverted it. Please take a look.